astropy/wcs/src/wcslib_wrap.c

```diff
@@ -3773,38 +3893,34 @@ PyWcsprm_set_velref(
   return set_int("velref", value, &self->x.velref);
 }
 
-/* static PyObject* */
-/* PyWcsprm_get_wtb( */
-/*     PyWcsprm* self, */
-/*     /\*@unused@*\/ void* closure) { */
 
-/*   PyObject* result; */
-/*   PyObject* subresult; */
-/*   int i, nwtb; */
+static PyObject* PyWcsprm_get_wtb(PyWcsprm* self, void* closure) {
+  PyObject* list;
+  PyObject* elem;
+  int i, nwtb;
 
-/*   nwtb = self->x.nwtb; */
+  nwtb = self->x.nwtb;
+
+  list = PyList_New(nwtb);
+  if (list == NULL) {
+    Py_DECREF(list);
+    return NULL;
+  }
 
-/*   result = PyList_New(nwtb); */
-/*   if (result == NULL) { */
-/*     return NULL; */
-/*   } */
+  for (i = 0; i < nwtb; ++i) {
+    elem = (PyObject *)PyWtbarr_cnew((PyObject *)self, &(self->x.wtb[i]));
+    if (elem == NULL) return NULL;
 
-/*   for (i = 0; i < nwtb; ++i) { */
-/*     subresult = (PyObject *)PyWtbarr_cnew((PyObject *)self, &(self->x.wtb[i])); */
-/*     if (subresult == NULL) { */
-/*       Py_DECREF(result); */
-/*       return NULL; */
-/*     } */
+    if (PyList_SetItem(list, i, elem) == -1) {
+      Py_DECREF(elem);
```

Comment:
`PyList_SetItem` steals a reference to `elem`, so it must not be decrefed again.

Reply:
But since we just created the list, you could use `PyList_SET_ITEM` here instead because we can garantuee that `elem` is not NULL and the `i` is within the bounds of the list.

Reply:
Latest commit will use `PyList_Append`.

Reply:
I don't know if that will work. I think the `PyList_New` sets the size (not only the allocated size) and Append would then append after these. So we would have a list containing NULLs which will probably crash Python.

There's nothing wrong with using `PyList_SetItem` or `PyList_SET_ITEM`. The nicest thing about `PyList_SET_ITEM` here is that we don't need to account for (not possible) failures.

Reply:
You are right. My tests did not catch this but I am working on a new set of tests that will catch this king of error.